apps/browser/src/vault/popup/components/vault/view.component.ts

```diff
@@ -141,6 +154,32 @@ export class ViewComponent extends BaseViewComponent {
   async load() {
     await super.load();
     await this.loadPageDetails();
+
+    switch (this.loadAction) {
+      case AUTOFILL_ID:
+        this.fillCipher();
+        break;
+      case COPY_USERNAME_ID:
+        await this.copy(this.cipher.login.username, "username", "Username");
+        if (this.inPopout) {
+          this.close();
+        }
+        break;
+      case COPY_PASSWORD_ID:
+        await this.copy(this.cipher.login.password, "password", "Password");
+        if (this.inPopout) {
+          this.close();
+        }
+        break;
+      case COPY_VERIFICATIONCODE_ID:
+        await this.copy(this.cipher.login.totp, "verificationCodeTotp", "TOTP");
+        if (this.inPopout) {
+          this.close();
+        }
+        break;
+      default:
+        break;
+    }
```

Comment:
```suggestion
    if (this.loadAction === AUTOFILL_ID) {
      this.fillCipher();
      return;
    }

    if (this.loadAction === COPY_USERNAME_ID) {
      await this.copy(this.cipher.login.username, "username", "Username");
    }

    if (this.loadAction === COPY_PASSWORD_ID) {
      await this.copy(this.cipher.login.password, "password", "Password");
    }

    if (this.loadAction === COPY_VERIFICATIONCODE_ID) {
      await this.copy(this.cipher.login.totp, "verificationCodeTotp", "TOTP");
    }

    if (this.inPopout) {
      this.close();
    }
```
Major Nitpick, feel free to disregard. I'm not against using `switch` statements, as often they can be more readable when compared to using single if statements...

Here though, given some of the repetition, it seems like you'd simplify the readability of the `load` method by structuring this using `if` statements instead...

Just a thought, not asking for a change here.

Reply:
Ah yeah - we don't even need that second load action array check. In my original code, without it the window would close immediately. After taking a beat, I realized I was doing a break from the `AUTOFILL_ID` case (which we don't await), so it was moving on to the `inPopout` check and closing prematurely.

Because `fillCipher` is handling the side-effects we care about elsewhere (unlike the copy actions), we can simply return from that case, forgoing the close altogether.